modules/ROOT/pages/traversal-framework/traversal_framework_java_api.adoc

```diff
@@ -0,0 +1,353 @@
+:description: The Neo4j Traversal Framework Java API.
+
+:org-neo4j-graphdb-Direction-both: {neo4j-javadocs-base-uri}/org/neo4j/graphdb/Direction.html#BOTH
+:org-neo4j-graphdb-traversal-description: {neo4j-javadocs-base-uri}/org/neo4j/graphdb/Transaction.html#traversalDescription()
+
+[[traversal-java-api]]
+= Traversal Framework Java API
+
+[[traversal-java-api-traversaldescription]]
+== link:{org-neo4j-graphdb-traversal-description}[TraversalDescription^]
+
+The `TraversalDescription` is the main interface used for defining and initializing traversals.
+It is not meant to be implemented by users of the Traversal Framework, but rather is provided as a way for the user to describe traversals.
+`TraversalDescription` instances are immutable and its methods return a new `TraversalDescription` that is modified compared to the object the method was invoked on with the arguments of the method.
+
+
+=== Relationships
+
+Adds a relationship type to the list of relationship types to traverse.
+By default, this list is empty meaning that it will traverse _all relationships_, regardless of type.
+If one or more relationships are present _only the added_ types will be traversed.
+There are two methods, one `including direction` and another `excluding direction`, where the latter traverses relationships in link:{org-neo4j-graphdb-Direction-both}[both directions^].
+
+
+[[traversal-java-api-evaluator]]
+== Evaluator
+
+An `Evaluator` takes the `Path` from the start node to the current node and decides whether:
```

Comment:
The traversal api gives a lot of control over what to do at each step, so potentially a user wants to check the whole path. This can cause memory issues with breadth first, I believe there is a comment about that somewhere in the docs as a warning already.

Reply:
I meant having a parameter you don't expect in what basis you should tune it is never a good idea. If I was a user of this I would be asking the same question I'm asking right now: how does this affect my algorithm and what's the recommended value?